src/settings.rs

```diff
@@ -157,11 +158,18 @@ impl Settings {
             },
             None => None,
         };
+        let profile = match get_opt_string(&mut table, "profile", path)? {
+            Some(profile) => match Profile::from_str(profile.as_str()) {
```

Comment:
this could use Result::ok() I think, no? 

``` let profile = get_opt_string(&mut table, "profile", path)?.map(|o| Profile::from_str(o.as_str()).ok());```

Reply:
It returns option<option<profile>>. But we want option<profile>.

Reply:
We should use `and_then`.

Reply:
Thanks for your review!